app/server.go

```diff
@@ -40,19 +41,46 @@ func Server(
 	return router
 }
 
+
+func failOnError(err error, msg string) {
+	if err != nil {
+		log.Fatalf("%s: %s", msg, err)
+	}
+}
+
 // Start runs the server
 func Start() {
 	pool, err := pgxpool.Connect(context.Background(), os.Getenv("DATABASE_URL"))
 	if err != nil {
 		log.Println(os.Getenv("DATABASE_URL"))
 		log.Fatalln("db failed", err)
 	}
+	conn, err := amqp.Dial("amqp://guest:guest@localhost:5672/")
+	failOnError(err, "can't connect")
+	defer conn.Close()
+
+	ch, err := conn.Channel()
+	failOnError(err, "failed to open channel")
+	defer ch.Close()
 
+	err = ch.ExchangeDeclare(
+		"profile", // name
+		"topic",      // type
+		true,         // durable
+		false,        // auto-deleted
+		false,        // internal
+		false,        // no-wait
+		nil,          // arguments
+	)
+	failOnError(err, "can't create exchange")
+	mm := usecase.NewMessagingManager(ch)
 	studentRepository := repository.NewStudentRepository(pool)
 	reviewRepository := repository4.NewReviewRepository(pool)
-	studentUseCase := usecase.NewStudentUseCase(studentRepository, reviewRepository, time.Second)
+	studentUseCase := usecase.NewStudentUseCase(mm, studentRepository, reviewRepository, time.Second)
 	studentHandler := http.NewStudentHandler(studentUseCase)
-
+	go studentUseCase.CreateStudentTopic()
```

Comment:
what is the purpose for sending (create, update, delete) messages here? I thought they would be called in the usecase or repo

Reply:
It's starts an infinite loop of sending async messages to rabbitmq. This method is called just once here, and no where else. Only one method run is needed for the entire app.